gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxSh.java

```diff
@@ -155,6 +170,81 @@ public boolean validate() {
     public abstract String getUsage();
   }
 
+  private class KnoxBuildTrustStore extends Command {
+
+    private static final String USAGE = "buildTrustStore --gateway server-url";
+    private static final String DESC = "Downloads the gateway server's public certificate and builds a trust store.";
+    private static final String CLIENT_TRUST_STORE_FILE_NAME = "gateway-client-trust.jks";
```

Comment:
Do we need to make the path configurable? Say there are multiple Knox instances? Also writing to "user.home" could overwrite a file?

Reply:
Hm...currently KnoxShell supports the following scenarios when building up KnoxSession:
truststore related attributes are either passed as environment variables (`KNOX_CLIENT_TRUSTSTORE_DIR`, `KNOX_CLIENT_TRUSTSTORE_FILENAME` and `KNOX_CLIENT_TRUSTSTORE_PASS`) or KnoxShell uses default values (`$USER_HOME`, `gateway-client-trust.jks` and `changeit`).
I can modify the new command to consider the same environment variables but I would rather enhance it by letting end-users to pass these attributes as optional program arguments.
If you agree I'd let this now as is and create a new JIRA to address this area. Any objection?

Reply:
I think we should make it consistent with where the KnoxShell will look for the truststore. Otherwise they could run the new command and then not be able to use it.

Probably makes sense to add helper methods to figure out what the correct paths are using the environment variables or the default values. I think passing into KnoxShell as optional parameters is probably overkill for this.

Reply:
Fine by me; will submit the new patch soon